src/twisted/names/dns.py

```diff
@@ -246,6 +249,38 @@ def _nameToLabels(name):
 
 
 
+def domainString(domain):
+    """
+    Coerce a domain name string to bytes.
+
+    L{twisted.names} represents domain names as L{bytes}, but many interfaces
+    accept L{bytes} or a text string (L{unicode} on Python 2, L{str} on Python
+    3). This function coerces text strings using IDNA encoding --- see
+    L{encodings.idna}.
+
+    Note that DNS is I{case insensitive} but I{case preserving}. This function
+    doesn't normalize case, so you'll still need to do that whenever comparing
+    the strings it returns.
+
+    @param domain: A domain name.  If passed as a text string it will be
+        C{idna} encoded.
+    @type domain: L{bytes} or L{str}
+
+    @returns: L{bytes} suitable for network transmission.
+    @rtype: L{bytes}
+
+    @since: Twisted NEXT
+    """
+    if isinstance(domain, unicode):
+        domain = domain.encode('idna')
```

Comment:
You might want to use `twisted/internet/_idna.py` here rather than a straight `.encode`.  (The python stdlib codec is a much older version of the spec.)

Reply:
Sure, will do.